mapbox/io/include/mapbox/io.hpp

```diff
@@ -15,8 +15,7 @@ namespace base {
 
 using ErrorType = std::string;
 
-// NOLINTNEXTLINE(misc-definitions-in-headers)
-nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
+inline nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
```

Comment:
maybe better to add definition in a .cpp file?

Reply:
I thought about it - but it'd break the header-only nature of those libraries, no?

Reply:
It's not a template class - imo does not have to be header-only.. 

Also, inline functions might bring some problems, e.g. someone could define
```
inline nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
// Unexpected leftover implementation here, or maybe previous implementation
}
```
Both implementations will build fine together, then arbitrary version will be picked for execution (if inline failed)..

Reply:
:point_right: https://github.com/mapbox/mapbox-base/pull/24